keras_cv/layers/preprocessing/random_resized_crop.py

```diff
@@ -50,103 +44,86 @@ class RandomResizedCrop(BaseImageAugmentationLayer):
-        self.area_factor = preprocessing.parse_factor(resize_factor,
-                                                      param_name="area_factor",
-                                                      seed=seed)
-        self.aspect_ratio_factor = preprocessing.parse_factor(
-            aspect_ratio_factor, param_name="aspect_ratio_factor", seed=seed)
-        if resize_factor == 0.0 and aspect_ratio_factor == 0.0:
-            warnings.warn(
-                "RandomResizedCrop received both `area_factor=0.0` and "
-                "`aspect_ratio_factor=0.0`. As a result, the layer will perform no "
-                "augmentation.")
+    def __init__(self,
+                 target_size,
+                 aspect_ratio_factor=(3. / 4., 4. / 3.),
+                 area_factor=(0.08, 1.0),
+                 interpolation="bilinear",
+                 fill_mode="reflect",
+                 fill_value=0.0,
+                 seed=None,
+                 **kwargs):
+        super(RandomResizedCrop, self).__init__(seed=seed, **kwargs)
+
+        self.target_size = target_size
+        self.aspect_ratio_factor = aspect_ratio_factor
+        self.area_factor = preprocessing.parse_factor(area_factor,
+            param_name="area_factor", seed=seed)
+
         self.interpolation = interpolation
         self.fill_mode = fill_mode
         self.fill_value = fill_value
         self.seed = seed
-        self.crop_size = crop_size
+
+        if area_factor == 0.0 and aspect_ratio_factor == 0.0:
+            warnings.warn(
+                "RandomResizedCrop received both `area_factor=0.0` and "
+                "`aspect_ratio_factor=0.0`. As a result, the layer will perform no "
+                "augmentation.")
 
     def get_random_transformation(self,
                                   image=None,
                                   label=None,
                                   bounding_box=None):
-        # random area and aspect ratio
-        random_area = (1.0 +
-                       preprocessing.random_inversion(self._random_generator) *
-                       self.area_factor())
-        random_aspect_ratio = (
-            1.0 + preprocessing.random_inversion(self._random_generator) *
-            self.aspect_ratio_factor())
-
-        # corresponding height and width (1 = original height/width)
-        new_height = tf.sqrt(random_area / random_aspect_ratio)
-        new_width = tf.sqrt(random_area * random_aspect_ratio)
-
-        # random offsets for the crop, inside or outside the image
-        height_offset = self._random_generator.random_uniform(
+        area_factor = self.area_factor()
+        aspect_ratio = tf.random.uniform((), minval=self.aspect_ratio_factor[0], 
+            maxval=self.aspect_ratio_factor[1], dtype=tf.float32)
+
+        new_height = tf.clip_by_value(tf.sqrt(area_factor / aspect_ratio), 0.0,
+                                    1.0)  # to avoid unwanted/unintuitive effects
```

Comment:
we should also be able to support zoomed in resizes- as in they can be larger than the original image